yt/geometry/particle_oct_container.pyx

```diff
@@ -1846,38 +1806,44 @@ cdef class ParticleBitmapSelector:
         for i in range(2):
             npos[0] = pos[0] + i*ndds[0]
             rpos[0] = npos[0] + ndds[0]
+            ncur_ind[0] = (cur_ind[0] << 1) + i
             for j in range(2):
                 npos[1] = pos[1] + j*ndds[1]
                 rpos[1] = npos[1] + ndds[1]
+                ncur_ind[1] = (cur_ind[1] << 1) + j
                 for k in range(2):
                     npos[2] = pos[2] + k*ndds[2]
                     rpos[2] = npos[2] + ndds[2]
+                    ncur_ind[2] = (cur_ind[2] << 1) + k
                     # Only recurse into selected cells
                     sbbox = self.selector.select_bbox_edge(npos, rpos)
-                    if sbbox == 0: continue
+                    if sbbox == 0:
+                        continue
                     if nlevel < self.order1:
                         if sbbox == 1:
-                            self.fill_subcells_mi1(npos, ndds)
+                            self.fill_subcells_mi1(nlevel, ncur_ind)
                         else:
-                            self.recursive_morton_mask(nlevel, npos, ndds, mi1)
+                            self.recursive_morton_mask(
+                                nlevel, npos, ndds, mi1, ncur_ind)
                     elif nlevel == self.order1:
-                        mi1 = bounded_morton_dds(npos[0], npos[1], npos[2], self.DLE, ndds)
+                        mi1 = encode_morton_64bit(
+                            ncur_ind[0], ncur_ind[1], ncur_ind[2])
                         if sbbox == 2: # an edge cell
                             if self.is_refined(mi1) == 1:
-                                self.recursive_morton_mask(nlevel, npos, ndds, mi1)
+                                self.recursive_morton_mask(
+                                    nlevel, npos, ndds, mi1, zeros)
```

Comment:
I think this is right, but it took me a second to work out that this was not mutated on the other end.

Reply:
yup! you did this in a nice clever way without mutating anything :)

Reply:
i guess this is worth a comment though